### PR TITLE
docs: update v9 migration doc for merkle path v2

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Migration docs for changes included in #6644 

needs: #6644
closes: #6496 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 1]
```diff
@@ -24,23 +24,54 @@ There are four sections based on the four potential user groups of this document
...
+The legacy function `RegisterInterchainAccount` now takes an extra parameter to specify the ordering of new ICA channels:
+
+```diff
+func (k Keeper) RegisterInterchainAccount(
+  ctx sdk.Context,
+  connectionID, owner,
+  version string,
++ ordering channeltypes.Order
+) error {
+```
```
Comment (damiannolan) [inline docs/docs/05-migrations/13-v8-to-v9.md:51]: I moved this section from core to apps. Sorry about the diff!

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 2]
```diff
@@ -24,23 +24,54 @@ There are four sections based on the four potential user groups of this document
...
+- The `exported.ConnectionI` and `exported.CounterpartyConnectionI` interfaces have been removed. Please use the concrete types.
+- The `exported.Proof` interface has been removed. Please use the `MerkleProof` concrete type.
+- The functions `GetState()`, `GetOrdering()`, `GetCounterparty()`, `GetConnectionHops()`, `GetVersion()` of the `Channel` type have been removed.
+- The functions `GetPortID()`, `GetChannelID()` of the `CounterpartyChannel` type have been removed.
+- The functions `GetClientID()`, `GetState()`, `GetCounterparty()`, `GetVersions()`, and `GetDelayPeriod` of the `Connection` type have been removed. 
+- The functions `GetClientID()`, `GetConnectionID()`, and `GetPrefix()` of the `CounterpartyConnection` type have been removed. 
+- The utility function `QueryLatestConsensusState` of `04-channel` CLI has been removed.
+- `UnmarshalPacketData` now takes in the context, portID, and channelID. This allows the packet data to be unmarshaled based on the channel version.
+- `Router` reference has been removed from IBC core keeper: [#6138](https://github.com/cosmos/ibc-go/pull/6138). Please use `PortKeeper.Router` instead.
```
Comment (damiannolan) [inline docs/docs/05-migrations/13-v8-to-v9.md:66]: Made bullets

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 3]
```diff
@@ -24,23 +24,54 @@ There are four sections based on the four potential user groups of this document
...
 
-The functions `GetState()`, `GetOrdering()`, `GetCounterparty()`, `GetConnectionHops()`, `GetVersion()` of the `Channel` type have been removed.
-The functions `GetPortID()`, `GetChannelID()` of the `CounterpartyChannel` type have been removed.
-The functions `GetClientID()`, `GetState()`, `GetCounterparty()`, `GetVersions()`, and `GetDelayPeriod` of the `Connection` type have been removed. 
-The functions `GetClientID()`, `GetConnectionID()`, and `GetPrefix()` of the `CounterpartyConnection` type have been removed. 
+### 02-client
 
-The utility function `QueryLatestConsensusState` of `04-channel` CLI has been removed.
+- The `QueryVerifyMembershipRequest` protobuf message has been modified to include `commitment.v2.MerklePath`. The deprecated `commitment.v1.MerklePath` field has been `reserved`. [See 23-commitment](#23-commitment)
```
Comment (damiannolan) [inline docs/docs/05-migrations/13-v8-to-v9.md:70]: Entry for change to `QueryVerifyMembershipRequest` msg

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 4]
```diff
@@ -24,23 +24,54 @@ There are four sections based on the four potential user groups of this document
...
 
-### API deprecation notice
+### 23-commitment
 
-The testing package functions `coordinator.Setup`, `coordinator.SetupClients`, `coordinator.SetupConnections`, `coordinator.CreateConnections`, and `coordinator.CreateChannels` have been deprecated and will be removed in v10.
-Please use the new functions `path.Setup`, `path.SetupClients`, `path.SetupConnections`, `path.CreateConnections`, `path.CreateChannels`.
+- The `MerklePath` type has been deprecated and a new `commitment.v2.MerklePath` type has been introduced (ref: [#6644](https://github.com/cosmos/ibc-go/pull/6644)). The new `commitment.v2.MerklePath` contains `repeated bytes` in favour of `repeated string`. This allows users to prove values stored under keys which contain non-utf8 encoded symbols. As a result, changes have been made the 02-client `Query` service and 08-wasm contract api messages for JSON blobs. See [02-client](#02-client) and [08-wasm](#08-wasm), respectively.
```
Comment (damiannolan) [inline docs/docs/05-migrations/13-v8-to-v9.md:74]: Entry for 23-commitment changes.

File: docs/docs/05-migrations/13-v8-to-v9.md [hunk 5]
```diff
@@ -139,5 +144,6 @@ The `IterateConsensusMetadata` function has been removed.
 
 ### 08-wasm
 
-The `ExportMetadataMsg` struct has been removed and is no longer required for contracts to implement. Core IBC will handle exporting all key/value's written to the store by a light client contract.
-The `ZeroCustomFields` interface function has been removed from the `ClientState` interface. Core IBC only used this function to set tendermint client states when scheduling an IBC software upgrade. The interface function has been replaced by a type assertion.  
+- The `VerifyMembershipMsg` and `VerifyNonMembershipMsg` payloads for `SudoMsg` have been extended to include a new field, `MerklePath`. The existing `Path` field will remain the same. The new `MerklePath` field is used if and only if the provided key path contains non-utf8 encoded symbols, and as a result will encode the JSON field `merkle_path` as a base64 encoded bytestring. See [23-commitment](#23-commitment).
```
Comment (damiannolan) [inline docs/docs/05-migrations/13-v8-to-v9.md:147]: Entry for 08-wasm contract api additions

**Review by damiannolan — COMMENTED** [review]
Left some notes! Sorry about the messy diff, but the doc needed to cleaned up a little

**Review by colin-axner — APPROVED** [review]
Looks great, thank you! :heart:

**Review by crodriguezvega — APPROVED** [review]
Looks great. Thank you, @damiannolan

**Pull request merged**